dbms/src/Interpreters/RowRefs.h

```diff
@@ -216,13 +216,17 @@ class AsofRowRefs
     using Lookups = std::variant<
         Entry<UInt32>::LookupPtr,
         Entry<UInt64>::LookupPtr,
+        Entry<Int32>::LookupPtr,
+        Entry<Int64>::LookupPtr,
         Entry<Float32>::LookupPtr,
         Entry<Float64>::LookupPtr>;
 
     enum class Type
     {
-        key32,
-        key64,
+        keyu32,
+        keyu64,
+        keyi32,
+        keyi64,
```

Comment:
I have not found ASOF JOIN test for this change

Reply:
Good point, I see that it was removed, so let me check...

Reply:
Fixed using DateTime64 in ASOF join and also updated tests, see 00927_asof_join_other_types